### PR TITLE
Setting up zk-proving, adding pubkeys for Sigmojis II, and fixing README

### PR DESCRIPTION
**Pull request by RiverRuby**
This PR adds [spartan-ecdsa](https://github.com/personaelabs/spartan-ecdsa) based ZK-proofs on top of signatures from the v1 of the ARX cards that just produce secp256k1 sigs. We will need to rewrite the proving code when we have baby jubjub sigs, but much of the same input processing and logic will be used.

Other small tasks include adding the public keys for the second set of Sigmoji cards. It also updates the README to be more descriptive.

### CODE REVIEW

File: components/modals/ProvingModal.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  ModalContainer,
+  ModalDiv,
+  ModalHeader,
+  OuterContainer,
+  InnerContainer,
+} from "../shared/Modal";
+import { ProverWasm, initWasm, makeProofs } from "@/lib/zkProving";
+
+export default function ProvingModal() {
+  const [wasm, setWasm] = useState<ProverWasm | null>(null);
```
Comment (AndrewCLu) [inline components/modals/ProvingModal.tsx]: Can just be `const [wasm, setWasm] = useState<ProverWasm>();`

File: lib/zkProving.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    enableProfiler: true,
+  });
+
+  await prover.initWasm();
+
+  return { poseidon, prover };
+}
+
+export const makeProofs = async (proverWasm: ProverWasm) => {
+  const treeDepth = 20;
```
Comment (AndrewCLu) [inline lib/zkProving.ts:38]: Do we need a tree this large?

File: lib/zkProving.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  const pubKeyTree = new Tree(treeDepth, proverWasm.poseidon);
+
+  for (const entry of Object.entries(cardPubKeys)) {
+    const secondaryPublicKeyRawUint8Array = new Uint8Array(
+      (entry[1].secondaryPublicKeyRaw.match(/.{1,2}/g) || []).map((byte) =>
+        parseInt(byte, 16)
```
Comment (AndrewCLu) [inline lib/zkProving.ts]: Would be useful to refactor this into utils, perhaps named `publicKeyHexToUint8Array`

File: lib/zkProving.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        )
+      );
+      const pubKey = importPublic(secondaryPublicKeyRawUint8Array);
+      const pubKeyBuffer = Buffer.from(pubKey);
+      const pubKeyHash = proverWasm.poseidon.hashPubKey(pubKeyBuffer);
+      const index = pubKeyTree.indexOf(pubKeyHash);
+      const merkleProof = pubKeyTree.createProof(index);
+
+      // set up msgHashArray
+      const rndBuf = Buffer.from(sigmoji.PCD.proof.signedDigest, "hex");
```
Comment (AndrewCLu) [inline lib/zkProving.ts:80]: Seems like this matches the `verify` code in the PCD definition. Might want to refactor this to utils to make it more clear

**Review by AndrewCLu — COMMENTED** [review]
Left a few comments, mostly about refactoring code for readability.

File: lib/zkProving.ts (hunk 2 above)
Comment (RiverRuby) [reply lib/zkProving.ts:38]: We don't, but that's just what the current spartan-ecdsa circuits provide for. We could recompile their circuits with a lower tree size (we only need depth 5) but I'm not sure if it's worth going through that effort given we are hoping to not use this proof past Berlin.

File: lib/zkProving.ts (hunk 4 above)
Comment (RiverRuby) [reply lib/zkProving.ts:80]: Fair point; I'll leave it here for now since it doesn't get used again.

File: lib/zkProving.ts (hunk 3 above)
Comment (RiverRuby) [reply lib/zkProving.ts]: Very good point, cleaned up code a lot!

**Pull request merged**